libdnf/dnf-repo.cpp

```diff
@@ -1160,6 +1163,20 @@ dnf_repo_setup(DnfRepo *repo, GError **error)
     priv->urlvars = lr_urlvars_set(priv->urlvars, "basearch", basearch);
     testdatadir = dnf_realpath(TESTDATADIR);
     priv->urlvars = lr_urlvars_set(priv->urlvars, "testdatadir", testdatadir);
+    yum_vars_dir = g_dir_open(yum_vars_path, 0, NULL); /* non-fatal */
+    if (yum_vars_dir != NULL) {
+        while ((var_name = g_dir_read_name(yum_vars_dir)) != NULL) {
+            g_autofree gchar *contents = NULL;
+            g_autofree gchar *var_path = g_build_filename(yum_vars_path, var_name, NULL);
+            if (g_file_test(var_path, G_FILE_TEST_IS_REGULAR)) {
+                if (!g_file_get_contents(var_path, &contents, NULL, error)) {
+                    return FALSE;
```

Comment:
Minor: Leaks `yum_vars_dir` on failure.  Use `g_autoptr(GDir) yum_vars_dir = NULL` above to fix this, and remove the explicit `g_dir_close`.

Otherwise...seems OK to me.

Reply:
OK, fixed that. Thanks for catching that!